lib/layoutTestingMain.js

```diff
@@ -0,0 +1,56 @@
+var layoutEngine = require('./layout/forceDirectedLayout');
+var parse = require('./parse/parse');
+
+var log = function(message) {
+  console.log(message);
+
+  var li = document.createElement('li');
+  li.innerHTML = message;
+  document.getElementById('output').appendChild(li);
+};
+
+var StopWatchLogger = function() {};
+StopWatchLogger.prototype = {
+  start: function(name) {
+    log('' + name + ' started');
+    this.name = name;
+    this.startTime = Date.now();
+  },
+
+  stop: function() {
+    log('' + this.name + ' completed [' + (Date.now() - this.startTime) + 'ms]');
+  }
+};
+
+function processGcl(gcl) {
+  var stopwatch = new StopWatchLogger();
+
+  stopwatch.start('Parsing');
+  var parseResult = parse(gcl);
+  stopwatch.stop();
+
+  var errors = parseResult.gclErrors;
+  console.log(errors);
+
+  var model = parseResult.model;
+
+  stopwatch.start('Layout engine');
+  layoutEngine(model);
+  stopwatch.stop();
+  console.debug(model);
+}
+
+gclSubmit = function() {
```

Comment:
Ah the problem really is a lack of a "dom ready" synthetic event during
which to bind an event handler. Since we have no libraries I think this is
the proper old-fashioned way to do it — something has to be global and
available to the HTML.

On Fri, Mar 13, 2015 at 10:43 AM Robert Gilmore <notifications@github.com>
wrote:

> In lib/layoutTestingMain.js
> <https://github.com/twosigma/goll-e/pull/29#discussion_r26390624>:
>
> > +  stopwatch.start('Parsing');
> > +  var parseResult = parse(gcl);
> > +  stopwatch.stop();
> > +
> > +  var errors = parseResult.gclErrors;
> > +  console.log(errors);
> > +
> > +  var model = parseResult.model;
> > +
> > +  stopwatch.start('Layout engine');
> > +  layoutEngine(model);
> > +  stopwatch.stop();
> > +  console.debug(model);
> > +}
> > +
> > +gclSubmit = function() {
>
> What sort of trouble? Did event.preventDefault not work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twosigma/goll-e/pull/29/files#r26390624>.
>